controllers/mlb_player.go

```diff
@@ -0,0 +1,68 @@
+package controllers
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/EloYaniel/academy-go-q42021/entities"
+	r "github.com/EloYaniel/academy-go-q42021/repositories/implementations"
+	"github.com/EloYaniel/academy-go-q42021/services"
+	"github.com/gorilla/mux"
+)
+
+type errorMessage struct {
+	Message string `json:"message"`
+}
+
+var service services.MLBPlayerService = *services.NewMLBPlayerService(&r.CSVMLBPlayerRepository{})
+
+func GetMLBPlayers(w http.ResponseWriter, r *http.Request) {
```

Comment:
I would recommend having these controller functions being implemented by a controller struct, and inside that struct having your `service` variable. 

And similarly to how your defined the repository to be an interface that you use within your service, you can do the same here, you can define the service as an interface and use its functions in these controller functions. That will make unit testing a lot easier

Reply:
By having the service as an attribute on your structure(as an interface), you would then probably need a "constructor" function just like the one you have to create a new Service